### PR TITLE
RHDEVDOCS 5979 starting pipeline runs manually and on chatops, support OWNERS fully

### PR DESCRIPTION
**Pull request by mramendi**
Version(s):

pipelines-docs-main only 

Issue:

RHDEVDOCS 5979

Link to docs preview:

https://75409--ocpdocs-pr.netlify.app/openshift-pipelines/latest/pac/managing-pipeline-runs-pac.html (first two subsections)

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by mramendi** [comment]
https://issues.redhat.com/browse/RHDEVDOCS-5979

**Comment by piyush-garg** [comment]
lgtm

File: modules/op-creating-pipeline-run-using-pipelines-as-code.adoc [hunk 1]
```diff
@@ -63,19 +64,44 @@ You can use the following example to match the `pipeline-push-on-main` pipeline
...
+  pipelinesascode.tekton.dev/on-comment: "^/merge-pr"
+# ...
+----
+
+The pipeline run starts only if the comment author meets one of the following requirements:
+
+* The author is the owner of the repository.
+* The author is a collaborator on the repository.
+* The author is a public member on the organization of the repository.
+* The comment author is listed in the `approvers` or `reviewers` section of the `OWNERS` file in the root of the repository, as defined in the https://www.kubernetes.dev/docs/guide/owners/[Kubernetes documentation]. {pac} supports the full specification for the `OWNERS` and `OWNERS_ALIASES` files.
```
Comment (enarha) [inline modules/op-creating-pipeline-run-using-pipelines-as-code.adoc:102]: Here where repeating that line below, I would put a note stating that "when the OWNERS file is using filters (https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md#filters), approvers and reviewers are only matched against the ".*" filter.

**Review by enarha — COMMENTED** [review]
Apart from one comment, the rest is LGTM.

File: modules/op-creating-pipeline-run-using-pipelines-as-code.adoc [hunk 2]
```diff
@@ -63,19 +64,44 @@ You can use the following example to match the `pipeline-push-on-main` pipeline
...
+apiVersion: tekton.dev/v1
+kind: PipelineRun
+metadata:
+  name: pipeline-comment
+annotations:
+  pipelinesascode.tekton.dev/on-comment: "^/merge-pr"
+# ...
+----
+
+The pipeline run starts only if the comment author meets one of the following requirements:
```
Comment (chmouel) [inline modules/op-creating-pipeline-run-using-pipelines-as-code.adoc:97]: we create a PipelineRun to start a Pipeline in general in tekton and Pac only really deals with pipelinerun so you may want to change that..

Comment (mramendi) [reply modules/op-creating-pipeline-run-using-pipelines-as-code.adoc:97]: I've made the changes I believe

File: modules/op-running-pipeline-run-using-pipelines-as-code.adoc [hunk 3]
```diff
@@ -13,15 +13,7 @@ In the event of a pull request or a merge request, {pac} also runs pipelines fro
...
-* The pull request author is listed in an `OWNER` file located in the repository root of the `main` branch as defined in the GitHub configuration for the repository. Also, the  pull request author is added to either `approvers` or `reviewers` section. For example, if an author is listed in the `approvers` section, then a pull request raised by that author starts the pipeline run.
-
-[source,yaml]
-----
-...
-  approvers:
-    - approved
-...
-----
+* The pull request author is listed in the `approvers` or `reviewers` section of the `OWNERS` file in the root of the repository, as defined in the https://www.kubernetes.dev/docs/guide/owners/[Kubernetes documentation]. {pac} supports the full specification for the `OWNERS` and `OWNERS_ALIASES` files.
```
Comment (chmouel) [inline modules/op-running-pipeline-run-using-pipelines-as-code.adoc:16]: I don't think we support the full specification but only a subset, @enarha would know more about which ones.

Comment (enarha) [reply modules/op-running-pipeline-run-using-pipelines-as-code.adoc:16]: Yea, the same sentence repeats few times. I commented on the first instance (https://github.com/openshift/openshift-docs/pull/75409#discussion_r1587186853), but it applies here as well.

Comment (enarha) [reply modules/op-running-pipeline-run-using-pipelines-as-code.adoc:16]: @mramendi the update above is good, just apply it here as well.

**Review by enarha — COMMENTED** [review]
/lgtm

**Comment by chmouel** [comment]
Lgtm

File: modules/op-creating-pipeline-run-using-pipelines-as-code.adoc [hunk 4]
```diff
@@ -59,23 +60,47 @@ You can use the following example to match the `pipeline-pr-main` pipeline with
...
-<1> You can specifiy multiple branches by adding comma-separated entries. For example, `"[main, release-nightly]"`. In addition, you can specify the following:
+<1> You can specify multiple branches by adding comma-separated entries. For example, `"[main, release-nightly]"`. In addition, you can specify the following:
 * Full references to branches such as `"refs/heads/main"`
 * Globs with pattern matching such as `"refs/heads/\*"`
 * Tags such as `"refs/tags/1.\*"`
 
+[discrete]
+.Matching a comment event to a pipeline run
+
+You can use the following example to match the `pipeline-comment` pipeline run with a comment on a pull request, when the text of the comment matches the `^/merge-pr` regular expression:
```
Comment (savitaashture) [inline modules/op-creating-pipeline-run-using-pipelines-as-code.adoc:84]: @chmouel we also support `on-comment` for push as well right ?

Comment (chmouel) [reply modules/op-creating-pipeline-run-using-pipelines-as-code.adoc:84]: @savitaashture no still waiting this to be merged 🙃 https://github.com/openshift-pipelines/pipelines-as-code/pull/1638 (would not make it for osp1.5 now as there is not enough time to dogfood this)

**Review by ppitonak — APPROVED** [review]

File: modules/op-creating-pipeline-run-using-pipelines-as-code.adoc (hunk 4 above)
Comment (savitaashture) [reply modules/op-creating-pipeline-run-using-pipelines-as-code.adoc:84]: Ah i see

**Review by kowen-rh — COMMENTED** [review]
This is very clean! I just see what looks like a very minor typo to me. :slightly_smiling_face: 

Other than that, LGTM! :+1:

/remove-label peer-review-in-progress
/label peer-review-done

File: modules/op-creating-pipeline-run-using-pipelines-as-code.adoc [hunk 5]
```diff
@@ -59,23 +60,47 @@ You can use the following example to match the `pipeline-pr-main` pipeline with
...
+# ...
 ----
-<1> You can specifiy multiple branches by adding comma-separated entries. For example, `"[main, release-nightly]"`. In addition, you can specify the following:
+<1> You can specify multiple branches by adding comma-separated entries. For example, `"[main, release-nightly]"`. In addition, you can specify the following:
 * Full references to branches such as `"refs/heads/main"`
 * Globs with pattern matching such as `"refs/heads/\*"`
 * Tags such as `"refs/tags/1.\*"`
 
+[discrete]
+.Matching a comment event to a pipeline run
```
Comment (chmouel) [inline modules/op-creating-pipeline-run-using-pipelines-as-code.adoc:82]: can we mark it a s TP ?

File: modules/op-restarting-or-canceling-pipeline-run-using-pipelines-as-code.adoc [hunk 6]
```diff
@@ -5,13 +5,13 @@
...
 
-You can restart or cancel a pipeline run with no events, such as sending a new commit to your branch or raising a pull request. To restart all pipeline runs, use the *Re-run all checks* feature in the GitHub App. 
+You can restart or cancel a pipeline run with no events, such as sending a new commit to your branch or raising a pull request. To restart all pipeline runs, use the *Re-run all checks* feature in the GitHub App.
 
 To restart all or specific pipeline runs, use the following comments:
 
 * The `/test` and `/retest` comment restarts all pipeline runs.
 
-* The `/test <pipeline_run_name>` and `/retest <pipeline_run_name>` comment restarts a specific pipeline run.
+* The `/test <pipeline_run_name>` and `/retest <pipeline_run_name>` comment starts or restarts a specific pipeline run. You can use this command to start any {pac} pipeline run on the repository, whether or not it was triggered by an event for this pipeline run.
```
Comment (chmouel) [inline modules/op-restarting-or-canceling-pipeline-run-using-pipelines-as-code.adoc:14]: that part is TP,

**Review by ppitonak — APPROVED** [review]
lgtm

**Review by sheriff-rh — APPROVED** [review]
Changes look good; merging.

**Pull request merged**